### ISSUE TITLE
🤖 [AUTO] Implement Strategy: random_forest_spy_daily_returns

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#447]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 4)
**Extracted:** 2025-12-25T02:45:36.354175

---

1.  **Strategy Name**: random_forest_spy_daily_returns
2.  **Strategy Type**: Regression, Directional (Mean-reverting tendencies due to predictors)
3.  **Timeframe**: 1D (Holding for one day, entering at close, exiting at next close)
4.  **Instruments**: SPY (S&P 500 ETF)

5.  **Entry Rules**:
    *   **Long Entry**: Enter a long position if the averaged predicted future one-day return from the random forest model (`retPred1`) is positive.
    *   **Short Entry**: Enter a short position if the averaged predicted future one-day return from the random forest model (`retPred1`) is negative.

6.  **Exit Rules**:
    *   **Take Profit**: Not explicitly defined, but implied by 1-day holding period.
    *   **Stop Loss**: Not explicitly defined.
    *   **Time-based Exit**: All positions are held for one trading day.

7.  **Indicators Required**:
    *   Previous 1-day return (`ret1`)
    *   Previous 2-day return (`ret2`)
    *   Previous 5-day return (`ret5`)
    *   Previous 20-day return (`ret20`)
    *   Random Forest (MATLAB `TreeBagger` function with `Method='regression'`, `MinLeaf=100`, and `NumPredictorsToSample` parameter defaulted to one-third of total predictors, which implies random forest functionality). The example used K=5 trees.

8.  **Risk Management**:
    *   **Position Sizing**: Fixed capital amount per trade.
    *   **Maximum Risk per Trade**: Not specified.
    *   **Risk-Reward Ratio**: Not specified.

9.  **Additional Notes**:
    *   This strategy utilizes bagging (bootstrapping training data) and random subspace (sampling predictors) to train an ensemble of regression trees, averaging their predictions to reduce overfitting.
    *   The training data was from Dec 22, 2004, to Sep 15, 2009. The test data was from Sep 16, 2009, to Jun 2, 2014.
    *   Performance: 7.2% CAGR (out-of-sample) and 0.5 Sharpe ratio (out-of-sample) for K=5 trees.
    *   Increasing K or setting `NumPredictorsToSample` to 'all' (pure bagging) was noted to degrade out-of-sample performance in the example.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/random_forest_spy_daily_returns.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
